packages/flutter_test/lib/src/finders.dart

```diff
@@ -365,13 +365,18 @@ class CommonFinders {
   ///
   /// ```dart
   /// expect(find.byTooltip('Back'), findsOneWidget);
+  /// expect(find.byTooltip(RegExp('Back.*')), findsNWidgets(2));
   /// ```
   ///
   /// If the `skipOffstage` argument is true (the default), then this skips
   /// nodes that are [Offstage] or that are from inactive [Route]s.
-  Finder byTooltip(String message, { bool skipOffstage = true }) {
+  Finder byTooltip(Pattern message, { bool skipOffstage = true }) {
     return byWidgetPredicate(
-      (Widget widget) => widget is Tooltip && widget.message == message,
+      (Widget widget) {
+        return widget is Tooltip && (message is RegExp
+            ? (message.hasMatch(widget.message ?? '') || message.hasMatch(widget.richMessage?.toPlainText() ?? ''))
```

Comment:
For the regex: Should we only match on the property that isn't null? In theory, your regex could match for empty string and in the case where richMessage is non-null, this would give you a positive result (because of the first condition) even though it may not actually match richMessage.

Reply:
Yeah, OK, I'll fix that.  I was assuming nobody would regex on and empty string, but I can see it happening.